onnxruntime/core/providers/cpu/controlflow/loop.cc

```diff
@@ -114,6 +114,8 @@ class LoopImpl {
 
   int64_t max_trip_count_;
   bool condition_;
+  size_t init_iter_num_rank_;
+  size_t init_condition_rank_;
 
```

Comment:
the 'init_' in these names seems superflous. the rank of each won't change will it?

Reply:
the rank can change. for e.g. like in the loop_test
https://github.com/microsoft/onnxruntime/blob/0e809ccc96e3e23a757d5e95cb268e5cefcc3735/onnxruntime/test/providers/cpu/controlflow/loop_test.cc#L290 

the initial condition can be a 0-d tensor. then in the loop condition is updated by output of Less operator, which is a 1-d tensor.